test/runtests.jl

```diff
@@ -381,15 +385,67 @@ end # @testset "Core Parsers.xparse"
+        ]
+            pos = 1
+            len = length(input)
+            local res
+            for expected in expected_vals
+                res = Parsers.xparse(Int, input, pos, len, xopts_with_groupmark)
+                @test res.val == expected
+                @test Parsers.ok(res.code)
+                pos += res.tlen
+            end
+            @test Parsers.ok(res.code)
+            @test Parsers.eof(res.code)
+        end
+
+        # Groupmark tests for floats
+        for (input, expected_vals) in [
+            ("1000,0000,2000,3000", (1000.0,0.0,2000.0,3000.0,)),
+            ("\"1000\",\"0000\",\"2000\",\"3000\"", (1000.0,0.0,2000.0,3000.0,)),
+            ("\"1$(g)0$(g)0$(g)0\",0000,\"2$(g)0$(g)0$(g)0\",3000", (1000.0,0.0,2000.0,3000.0,)),
+            ("1000,\"0$(g)0$(g)0$(g)0\",2000,\"3$(g)0$(g)0$(g)0\"", (1000.0,0.0,2000.0,3000.0,)),
+            ("1000.00,0000.00,2000.00,3000.00", (1000.0,0.0,2000.0,3000.0,)),
+            ("\"1000.00\",\"0000.00\",\"2000.00\",\"3000.00\"", (1000.0,0.0,2000.0,3000.0,)),
+            ("\"1$(g)0$(g)0$(g)0.00\",0000.00,\"2$(g)0$(g)0$(g)0.00\",3000.00", (1000.0,0.0,2000.0,3000.0,)),
+            ("1000,\"0$(g)0$(g)0$(g)0.00\",2000.00,\"3$(g)0$(g)0$(g)0.00\"", (1000.0,0.0,2000.0,3000.0,)),
+            ("1000.00e0,0000.00e0,2000.00e0,3000.00e0", (1000.0,0.0,2000.0,3000.0,)),
+            ("\"1000.00e0\",\"0000.00e0\",\"2000.00e0\",\"3000.00e0\"", (1000.0,0.0,2000.0,3000.0,)),
+            ("\"1$(g)0$(g)0$(g)0.00e0\",0000.00e0,\"2$(g)0$(g)0$(g)0.00e0\",3000.00e0", (1000.0,0.0,2000.0,3000.0,)),
+            ("1000,\"0$(g)0$(g)0$(g)0.00e0\",2000.00e0,\"3$(g)0$(g)0$(g)0.00e0\"", (1000.0,0.0,2000.0,3000.0,)),
+            ("1000e0,0000e0,2000e0,3000e0", (1000.0,0.0,2000.0,3000.0,)),
+            ("\"1000e0\",\"0000e0\",\"2000e0\",\"3000e0\"", (1000.0,0.0,2000.0,3000.0,)),
+            ("\"1$(g)0$(g)0$(g)0e0\",0000e0,\"2$(g)0$(g)0$(g)0e0\",3000e0", (1000.0,0.0,2000.0,3000.0,)),
+            ("1000,\"0$(g)0$(g)0$(g)0e0\",2000e0,\"3$(g)0$(g)0$(g)0e0\"", (1000.0,0.0,2000.0,3000.0,)),
+        ]
+            pos = 1
+            len = length(input)
+            local res
+            for expected in expected_vals
+                res = Parsers.xparse(Float64, input, pos, len, xopts_with_groupmark)
+                @test res.val == expected
+                @test Parsers.ok(res.code)
+                pos += res.tlen
+            end
+            @test Parsers.ok(res.code)
+            @test Parsers.eof(res.code)
+        end
+    end
+
+
     # #168
-    res = Parsers.xparse(Int, "1,729"; groupmark=',')
-    @test res.code == (OK | EOF)
-    @test res.tlen == 5
-    @test res.val == 1729
-    res = Parsers.xparse(Int, "1,729"; groupmark=UInt8(','))
-    @test res.code == (OK | EOF)
-    @test res.tlen == 5
-    @test res.val == 1729
+    res = Parsers.parse(Int, "1,729", Parsers._get_default_options(groupmark=UInt8(',')))
+    @test res == 1729
```

Comment:
One tricky thing to consider -- the main `Parsers.Options` constructor that handles `kwargs` and validates them, has its defaults set for `xparse`, not `xparse2` (by default `xparse` uses the `Parsers.XOPTIONS` const object, while `xparse2` uses `Parsers.OPTIONS `). This means that if the user wants to modify the behavior of `parse`/`tryparse`/`xparse2`, by providing their own Options object, they'd get suboptimal defaults (in this code block I'm `Parsers._get_default_options` function which has the right defaults). Notably, the default constructor sets `delim=','` while `xparse2` would want `delim=nothing`.

Reply:
good spot -- can you open an issue about that please?

Reply:
thanks! https://github.com/JuliaData/Parsers.jl/issues/184